packages/react-components/src/box/src/Box.tsx

```diff
@@ -0,0 +1,46 @@
+import { ComponentProps, ElementType, ForwardedRef, ReactElement, ReactNode } from "react";
+import { forwardRef, omitProps } from "../../shared";
+
+export interface InnerBoxProps {
+    /**
+    * An HTML element type or a custom React element type to render as.
+    */
+    as?: ElementType;
+    /**
+     * [Slot](?path=/docs/getting-started-slots--page) to render into.
+     */
+    slot?: string;
+    /**
+     * React children
+    */
+    children?: ReactNode;
+    /**
+    * @ignore
+    */
+    forwardedRef: ForwardedRef<any>;
+}
+
+export function InnerBox(props: InnerBoxProps): ReactElement {
+    const {
+        as: Wrapper = "div",
+        children,
+        forwardedRef,
+        ...rest
+    } = omitProps(props, ["slot"]);
+
+    return (
+        <Wrapper
+            {...rest}
+            ref={forwardedRef}
+        >
+            {children}
+        </Wrapper>
+    );
+}
+export const Box = forwardRef<InnerBoxProps>((props, ref) => (
+    <InnerBox {...props} forwardedRef={ref} />
+));
+
+export type BoxProps = ComponentProps<typeof Box>
```

Comment:
I feel like this might be the only line in component codes you might dislike. 

The idea here is that some component might require accès to a component props, either to extend them or to type an object that will be passed as props.

There is two choice available, either we export the props like this all the time, or if someone needs the box component's properties, he uses `ComponentProps<typeof Box>` every time.

I'd rather export the type by default everywhere